### PR TITLE
JAVA-2480: Upgrade Jackson to 2.10.0

### PR DESCRIPTION
**Pull request by emerkle826**

### CODE REVIEW

File: examples/src/main/java/com/datastax/oss/driver/examples/json/codecs/JacksonJsonCodec.java [hunk 1]
```diff
@@ -52,28 +52,28 @@
...
 
   /**
    * Creates a new instance for the provided {@code javaClass}, and using the provided {@link
-   * ObjectMapper}.
+   * JsonMapper}.
    *
    * @param javaClass the Java class this codec maps to.
    */
-  public JacksonJsonCodec(Class<T> javaClass, ObjectMapper objectMapper) {
+  public JacksonJsonCodec(Class<T> javaClass, JsonMapper jsonMapper) {
```
Comment (emerkle826) [inline examples/src/main/java/com/datastax/oss/driver/examples/json/codecs/JacksonJsonCodec.java:74]: I know this is a public method signature change, but it is in the `examples` project, so I figured it's not a big deal.

File: pom.xml [hunk 2]
```diff
@@ -69,8 +69,7 @@
     <hk2.version>2.5.0</hk2.version>
     <jax-rs.version>2.0.1</jax-rs.version>
     <jsr353-ri.version>1.1.4</jsr353-ri.version>
-    <jackson.version>2.9.9</jackson.version>
-    <jackson-databind.version>2.9.9.3</jackson-databind.version>
```
Comment (absurdfarce) [inline pom.xml:73]: It _looks like_ this property is referenced specifically in com.datastax.oss.driver.osgi.BundleOptions::jacksonBundles()... is that a problem?

Comment (emerkle826) [reply pom.xml:73]: good catch, I thought I caught all the references.

Comment (emerkle826) [reply pom.xml:73]: @adutra It looks like we separated the Jackson version from the Jackson databind version with this commit:
https://github.com/datastax/java-driver/commit/f6980df6e065bca4ed2db28297d56e8f55c8d514
Do you recall why? And do you think it's ok to upgrade to 2.10.0 for both?

Comment (adutra) [reply pom.xml:73]: Yes because versions for these two artifacts started diverging a while ago.

Comment (adutra) [reply pom.xml:73]: I think it might be easier to just keep declaring these 2 properties, even if they have the same value.

**Comment by adutra** [comment]
Pushed a small fix for the databind property and confirmed that the OSGi tests pass.

**Review by adutra — APPROVED** [review]
I will open an equivalent PR for the DSE driver.

**Pull request merged**